protocol/x/clob/e2e/long_term_orders_test.go

```diff
@@ -187,3 +187,49 @@ func TestCancelStatefulOrder(t *testing.T) {
 		})
 	}
 }
+
+func TestImmediateExecutionLongTermOrders(t *testing.T) {
+	tApp := testapp.NewTestAppBuilder().WithTesting(t).Build()
+	ctx := tApp.InitChain()
+
+	// Reject long-term IOC in CheckTx
+	for _, checkTx := range testapp.MustMakeCheckTxsWithClobMsg(
+		ctx,
+		tApp.App,
+		*clobtypes.NewMsgPlaceOrder(
+			constants.LongTermOrder_Carl_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_IOC,
+		),
+	) {
+		resp := tApp.CheckTx(checkTx)
+		require.Conditionf(t, resp.IsErr, "Expected CheckTx to error. Response: %+v", resp)
+		require.Contains(t, resp.Log, clobtypes.ErrStatefulOrdersCannotRequireImmediateExecution.Error())
+	}
+
+	// Reject long-term FOK in CheckTx
+	for _, checkTx := range testapp.MustMakeCheckTxsWithClobMsg(
+		ctx,
+		tApp.App,
+		*clobtypes.NewMsgPlaceOrder(
+			constants.LongTermOrder_Carl_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_FOK,
+		),
+	) {
+		resp := tApp.CheckTx(checkTx)
+		require.Conditionf(t, resp.IsErr, "Expected CheckTx to error. Response: %+v", resp)
+		require.Contains(t, resp.Log, clobtypes.ErrStatefulOrdersCannotRequireImmediateExecution.Error())
+	}
+
+	// Reject long-term IOC/FOK in DeliverTx
+	blockAdvancement := testapp.BlockAdvancementWithErrors{
+		BlockAdvancement: testapp.BlockAdvancement{
+			StatefulOrders: []clobtypes.Order{
+				constants.LongTermOrder_Carl_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_IOC,
+				constants.LongTermOrder_Carl_Num0_Id0_Clob0_Sell1BTC_Price50000_GTBT10_FOK,
+			},
+		},
+		ExpectedDeliverTxErrors: map[int]string{
+			0: clobtypes.ErrStatefulOrdersCannotRequireImmediateExecution.Error(),
```

Comment:
Can we rename [this error](https://github.com/dydxprotocol/v4-chain/blob/8f43054603ed7595b49cef40ba8dd38aa708c3de/protocol/x/clob/types/errors.go#L350) to `ErrLongTermOrdersCannotRequireImmediateExecution`?

Reply:
yep will do